plugins/generator-1.18.2/forge-1.18.2/templates/dispensebehaviour.java.ftl

```diff
@@ -35,34 +35,39 @@
 
 package ${package}.item.extension;
 
+<#compress>
 @Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD) public class ${name}ItemExtension {
     	@SubscribeEvent public static void init(FMLCommonSetupEvent event) {
     		event.enqueueWork(() -> DispenserBlock.registerBehavior(${mappedMCItemToItem(data.item)}, new OptionalDispenseItemBehavior() {
     			public ItemStack execute(BlockSource blockSource, ItemStack stack) {
+    				<#assign hasSuccessCondition = hasProcedure(data.dispenseSuccessCondition)>
     				ItemStack itemstack = stack.copy();
     				Level world = blockSource.getLevel();
     				Direction direction = blockSource.getBlockState().getValue(DispenserBlock.FACING);
     				int x = blockSource.getPos().getX();
     				int y = blockSource.getPos().getY();
     				int z = blockSource.getPos().getZ();
 
+    				<#if hasSuccessCondition>
     				this.setSuccess(<@procedureOBJToConditionCode data.dispenseSuccessCondition/>);
+    				</#if>
 
     				<#if hasProcedure(data.dispenseResultItemstack)>
     					boolean success = this.isSuccess();
     					<#if hasReturnValueOf(data.dispenseResultItemstack, "itemstack")>
-
-    						return <@procedureOBJToItemstackCode data.dispenseResultItemstack/>;
+    						return <@procedureOBJToItemstackCode data.dispenseResultItemstack, false/>;
     					<#else>
     						<@procedureOBJToCode data.dispenseResultItemstack/>
-    						if(success) itemstack.shrink(1);
+    						<#if hasSuccessCondition>if(success)</#if>
+    						itemstack.shrink(1);

```

Comment:
`itemstack.shrink(1);` will now be called in all cases (when hasSuccessCondition is absent), where before it was only called on success.

I think this is more right than before, but just letting this out to make sure it was intentional or not?

Reply:
If there's no success condition, `success` would always be true, so it's the same behavior as before

Reply:
Oh, I see